### PR TITLE
Fix regex unicode error

### LINKED ISSUE #1399: install breaks with unicode file names
Installing on Windows, 32-bit Python with with tar.gz or .whl files fails for unicode file names. Note that the wheel does build successfully when installing from tar.gz.

> pip install tinamit --no-index --find-links=./dist
Looking in links: ./dist
Collecting tinamit
Requirement already satisfied: numpy in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (1.14.2+mkl)
Requirement already satisfied: scipy in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (1.0.1)
Requirement already satisfied: matplotlib in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (2.2.2)
Requirement already satisfied: pillow in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (5.0.0)
Requirement already satisfied: pandas in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (0.23.0)
Requirement already satisfied: theano in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (1.0.1)
Requirement already satisfied: pymc3>=3.4 in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (3.4.1)
Requirement already satisfied: python_dateutil in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (2.7.2)
Requirement already satisfied: regex in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (2018.2.21)
Requirement already satisfied: lark-parser in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (0.5.6)
Requirement already satisfied: Babel in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (2.5.3)
Requirement already satisfied: pint in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (0.8.1)
Requirement already satisfied: pyshp in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (1.2.12)
Requirement already satisfied: taqdir in c:\users\jmalar1\pycharmprojects\taqdir (from tinamit) (0.1.0)
Requirement already satisfied: pysd in c:\users\jmalar1\pycharmprojects\pysd (from tinamit) (0.8.3)
Requirement already satisfied: lxml in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (4.2.1)
Requirement already satisfied: spotpy in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from tinamit) (1.3.27)
Requirement already satisfied: cycler>=0.10 in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from matplotlib->tinamit) (0.10.0)
Requirement already satisfied: pytz in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from matplotlib->tinamit) (2018.3)
Requirement already satisfied: kiwisolver>=1.0.1 in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from matplotlib->tinamit) (1.0.1)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from matplotlib->tinamit) (2.2.0)
Requirement already satisfied: six>=1.10 in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from matplotlib->tinamit) (1.11.0)
Requirement already satisfied: patsy>=0.4.0 in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pymc3>=3.4->tinamit) (0.5.0)
Requirement already satisfied: h5py>=2.7.0 in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pymc3>=3.4->tinamit) (2.7.1)
Requirement already satisfied: joblib>=0.9 in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pymc3>=3.4->tinamit) (0.11)
Requirement already satisfied: tqdm>=4.8.4 in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pymc3>=3.4->tinamit) (4.19.9)
Requirement already satisfied: parsimonious in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pysd->tinamit) (0.8.0)
Requirement already satisfied: yapf in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pysd->tinamit) (0.20.2)
Requirement already satisfied: xarray in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pysd->tinamit) (0.10.1)
Requirement already satisfied: xlwt in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pysd->tinamit) (1.3.0)
Requirement already satisfied: funcsigs in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pysd->tinamit) (1.0.2)
Requirement already satisfied: pydoe in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pysd->tinamit) (0.3.8)
Requirement already satisfied: xlrd in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from pysd->tinamit) (1.1.0)
Requirement already satisfied: setuptools in c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages (from kiwisolver>=1.0.1->matplotlib->tinamit) (39.2.0)
Building wheels for collected packages: tinamit
  Running setup.py bdist_wheel for tinamit ... error
  Complete output from command c:\users\jmalar1\appdata\local\programs\python\python36-32\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\jmalar1\\AppData\\Local\\Temp\\pip-install-imnf4lpi\\tinamit\\setup.py';f=getattr(tokenize, 'open', open)(__
file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d C:\Users\jmalar1\AppData\Local\Temp\pip-wheel-jv256esd --python-tag cp36:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib
  creating build\lib\pruebas
  copying pruebas\test_bf.py -> build\lib\pruebas
  copying pruebas\test_calib_valid_modelos.py -> build\lib\pruebas
  copying pruebas\test_conectado.py -> build\lib\pruebas
  copying pruebas\test_datos.py -> build\lib\pruebas
  copying pruebas\test_envolturas_bf.py -> build\lib\pruebas
  copying pruebas\test_geog.py -> build\lib\pruebas
  copying pruebas\test_mds.py -> build\lib\pruebas
  copying pruebas\test_microcalib.py -> build\lib\pruebas
  copying pruebas\test_números.py -> build\lib\pruebas
  copying pruebas\test_simuls_con_datos.py -> build\lib\pruebas
  copying pruebas\test_unidades.py -> build\lib\pruebas
  copying pruebas\__init__.py -> build\lib\pruebas
  creating build\lib\tinamit
  copying tinamit\BF.py -> build\lib\tinamit
  copying tinamit\Conectado.py -> build\lib\tinamit
  copying tinamit\MDS.py -> build\lib\tinamit
  copying tinamit\Modelo.py -> build\lib\tinamit
  copying tinamit\__init__.py -> build\lib\tinamit
  creating build\lib\tinamit\Análisis
  copying tinamit\Análisis\Calibs.py -> build\lib\tinamit\Análisis
  copying tinamit\Análisis\Datos.py -> build\lib\tinamit\Análisis
  copying tinamit\Análisis\Números.py -> build\lib\tinamit\Análisis
  copying tinamit\Análisis\sintaxis.py -> build\lib\tinamit\Análisis
  copying tinamit\Análisis\Valids.py -> build\lib\tinamit\Análisis
  copying tinamit\Análisis\__init__.py -> build\lib\tinamit\Análisis
  creating build\lib\tinamit\EnvolturasBF
  copying tinamit\EnvolturasBF\__init__.py -> build\lib\tinamit\EnvolturasBF
  creating build\lib\tinamit\EnvolturasMDS
  copying tinamit\EnvolturasMDS\PySD.py -> build\lib\tinamit\EnvolturasMDS
  copying tinamit\EnvolturasMDS\Vensim.py -> build\lib\tinamit\EnvolturasMDS
  copying tinamit\EnvolturasMDS\__init__.py -> build\lib\tinamit\EnvolturasMDS
  creating build\lib\tinamit\Geog
  copying tinamit\Geog\Geog.py -> build\lib\tinamit\Geog
  copying tinamit\Geog\__init__.py -> build\lib\tinamit\Geog
  creating build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\Animaciones.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\Arte.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\Botones.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\Cajas.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\CajasGenéricas.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\CajasSubEtapas.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\Controles.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\ControlesGenéricos.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\Formatos.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\IGU.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\Listas.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\Traducciones.py -> build\lib\tinamit\Interfaz
  copying tinamit\Interfaz\__init__.py -> build\lib\tinamit\Interfaz
  creating build\lib\tinamit\trads
  copying tinamit\trads\__init__.py -> build\lib\tinamit\trads
  creating build\lib\tinamit\Unidades
  copying tinamit\Unidades\conv.py -> build\lib\tinamit\Unidades
  copying tinamit\Unidades\trads.py -> build\lib\tinamit\Unidades
  copying tinamit\Unidades\__init__.py -> build\lib\tinamit\Unidades
  creating build\lib\tinamit\EnvolturasBF\DSSAT
  copying tinamit\EnvolturasBF\DSSAT\envoltDSSAT.py -> build\lib\tinamit\EnvolturasBF\DSSAT
  copying tinamit\EnvolturasBF\DSSAT\envoltDSSAT_senc.py -> build\lib\tinamit\EnvolturasBF\DSSAT
  copying tinamit\EnvolturasBF\DSSAT\Lector.py -> build\lib\tinamit\EnvolturasBF\DSSAT
  copying tinamit\EnvolturasBF\DSSAT\__init__.py -> build\lib\tinamit\EnvolturasBF\DSSAT
  creating build\lib\tinamit\EnvolturasBF\SAHYSMOD
  copying tinamit\EnvolturasBF\SAHYSMOD\envoltura.py -> build\lib\tinamit\EnvolturasBF\SAHYSMOD
  copying tinamit\EnvolturasBF\SAHYSMOD\_anlzd_sntct.py -> build\lib\tinamit\EnvolturasBF\SAHYSMOD
  copying tinamit\EnvolturasBF\SAHYSMOD\_sahysmodIE.py -> build\lib\tinamit\EnvolturasBF\SAHYSMOD
  copying tinamit\EnvolturasBF\SAHYSMOD\__init__.py -> build\lib\tinamit\EnvolturasBF\SAHYSMOD
  creating build\lib\tinamit\EnvolturasBF\Tikon
  copying tinamit\EnvolturasBF\Tikon\envoltTikon.py -> build\lib\tinamit\EnvolturasBF\Tikon
  copying tinamit\EnvolturasBF\Tikon\__init__.py -> build\lib\tinamit\EnvolturasBF\Tikon
  creating build\lib\tinamit\Interfaz\Imágenes
  copying tinamit\Interfaz\Imágenes\__init__.py -> build\lib\tinamit\Interfaz\Imágenes
  creating build\lib\tinamit\trads\nl
  copying tinamit\trads\nl\BF.py -> build\lib\tinamit\trads\nl
  copying tinamit\trads\nl\Conectado.py -> build\lib\tinamit\trads\nl
  copying tinamit\trads\nl\definiciones.py -> build\lib\tinamit\trads\nl
  copying tinamit\trads\nl\MDS.py -> build\lib\tinamit\trads\nl
  copying tinamit\trads\nl\Modelo.py -> build\lib\tinamit\trads\nl
  copying tinamit\trads\nl\__init__.py -> build\lib\tinamit\trads\nl
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\jmalar1\AppData\Local\Temp\pip-install-imnf4lpi\tinamit\setup.py", line 179, in <module>
      'sdist': dist_f  # Comanda para crear distribución instalable
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages\setuptools\__init__.py", line 129, in setup
      return distutils.core.setup(**attrs)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\core.py", line 148, in setup
      dist.run_commands()
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dist.py", line 955, in run_commands
      self.run_command(cmd)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dist.py", line 974, in run_command
      cmd_obj.run()
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages\wheel\bdist_wheel.py", line 202, in run
      self.run_command('build')
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dist.py", line 974, in run_command
      cmd_obj.run()
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\command\build.py", line 135, in run
      self.run_command(cmd_name)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dist.py", line 974, in run_command
      cmd_obj.run()
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages\setuptools\command\build_py.py", line 52, in run
      self.build_packages()
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\command\build_py.py", line 372, in build_packages
      self.build_module(module, module_file, package)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages\setuptools\command\build_py.py", line 75, in build_module
      package)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\command\build_py.py", line 342, in build_module
      self.mkpath(dir)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\cmd.py", line 338, in mkpath
      dir_util.mkpath(name, mode, dry_run=self.dry_run)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dir_util.py", line 66, in mkpath
      log.info("creating %s", head)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\log.py", line 44, in info
      self._log(INFO, msg, args)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\log.py", line 34, in _log
      stream.write('%s\n' % msg)
    File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\encodings\cp1252.py", line 19, in encode
      return codecs.charmap_encode(input,self.errors,encoding_table)[0]
  UnicodeEncodeError: 'charmap' codec can't encode characters in position 33-34: character maps to <undefined>

  ----------------------------------------
  Failed building wheel for tinamit
  Running setup.py clean for tinamit
Failed to build tinamit
recommonmark 0.4.0 has requirement commonmark<=0.5.4, but you'll have commonmark 0.7.4 which is incompatible.
Installing collected packages: tinamit
  Running setup.py install for tinamit ... error
    Complete output from command c:\users\jmalar1\appdata\local\programs\python\python36-32\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\jmalar1\\AppData\\Local\\Temp\\pip-install-imnf4lpi\\tinamit\\setup.py';f=getattr(tokenize, 'open', open)(
__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\jmalar1\AppData\Local\Temp\pip-record-0q2wrmex\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib
    creating build\lib\pruebas
    copying pruebas\test_bf.py -> build\lib\pruebas
    copying pruebas\test_calib_valid_modelos.py -> build\lib\pruebas
    copying pruebas\test_conectado.py -> build\lib\pruebas
    copying pruebas\test_datos.py -> build\lib\pruebas
    copying pruebas\test_envolturas_bf.py -> build\lib\pruebas
    copying pruebas\test_geog.py -> build\lib\pruebas
    copying pruebas\test_mds.py -> build\lib\pruebas
    copying pruebas\test_microcalib.py -> build\lib\pruebas
    copying pruebas\test_números.py -> build\lib\pruebas
    copying pruebas\test_simuls_con_datos.py -> build\lib\pruebas
    copying pruebas\test_unidades.py -> build\lib\pruebas
    copying pruebas\__init__.py -> build\lib\pruebas
    creating build\lib\tinamit
    copying tinamit\BF.py -> build\lib\tinamit
    copying tinamit\Conectado.py -> build\lib\tinamit
    copying tinamit\MDS.py -> build\lib\tinamit
    copying tinamit\Modelo.py -> build\lib\tinamit
    copying tinamit\__init__.py -> build\lib\tinamit
    creating build\lib\tinamit\Análisis
    copying tinamit\Análisis\Calibs.py -> build\lib\tinamit\Análisis
    copying tinamit\Análisis\Datos.py -> build\lib\tinamit\Análisis
    copying tinamit\Análisis\Números.py -> build\lib\tinamit\Análisis
    copying tinamit\Análisis\sintaxis.py -> build\lib\tinamit\Análisis
    copying tinamit\Análisis\Valids.py -> build\lib\tinamit\Análisis
    copying tinamit\Análisis\__init__.py -> build\lib\tinamit\Análisis
    creating build\lib\tinamit\EnvolturasBF
    copying tinamit\EnvolturasBF\__init__.py -> build\lib\tinamit\EnvolturasBF
    creating build\lib\tinamit\EnvolturasMDS
    copying tinamit\EnvolturasMDS\PySD.py -> build\lib\tinamit\EnvolturasMDS
    copying tinamit\EnvolturasMDS\Vensim.py -> build\lib\tinamit\EnvolturasMDS
    copying tinamit\EnvolturasMDS\__init__.py -> build\lib\tinamit\EnvolturasMDS
    creating build\lib\tinamit\Geog
    copying tinamit\Geog\Geog.py -> build\lib\tinamit\Geog
    copying tinamit\Geog\__init__.py -> build\lib\tinamit\Geog
    creating build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\Animaciones.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\Arte.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\Botones.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\Cajas.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\CajasGenéricas.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\CajasSubEtapas.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\Controles.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\ControlesGenéricos.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\Formatos.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\IGU.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\Listas.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\Traducciones.py -> build\lib\tinamit\Interfaz
    copying tinamit\Interfaz\__init__.py -> build\lib\tinamit\Interfaz
    creating build\lib\tinamit\trads
    copying tinamit\trads\__init__.py -> build\lib\tinamit\trads
    creating build\lib\tinamit\Unidades
    copying tinamit\Unidades\conv.py -> build\lib\tinamit\Unidades
    copying tinamit\Unidades\trads.py -> build\lib\tinamit\Unidades
    copying tinamit\Unidades\__init__.py -> build\lib\tinamit\Unidades
    creating build\lib\tinamit\EnvolturasBF\DSSAT
    copying tinamit\EnvolturasBF\DSSAT\envoltDSSAT.py -> build\lib\tinamit\EnvolturasBF\DSSAT
    copying tinamit\EnvolturasBF\DSSAT\envoltDSSAT_senc.py -> build\lib\tinamit\EnvolturasBF\DSSAT
    copying tinamit\EnvolturasBF\DSSAT\Lector.py -> build\lib\tinamit\EnvolturasBF\DSSAT
    copying tinamit\EnvolturasBF\DSSAT\__init__.py -> build\lib\tinamit\EnvolturasBF\DSSAT
    creating build\lib\tinamit\EnvolturasBF\SAHYSMOD
    copying tinamit\EnvolturasBF\SAHYSMOD\envoltura.py -> build\lib\tinamit\EnvolturasBF\SAHYSMOD
    copying tinamit\EnvolturasBF\SAHYSMOD\_anlzd_sntct.py -> build\lib\tinamit\EnvolturasBF\SAHYSMOD
    copying tinamit\EnvolturasBF\SAHYSMOD\_sahysmodIE.py -> build\lib\tinamit\EnvolturasBF\SAHYSMOD
    copying tinamit\EnvolturasBF\SAHYSMOD\__init__.py -> build\lib\tinamit\EnvolturasBF\SAHYSMOD
    creating build\lib\tinamit\EnvolturasBF\Tikon
    copying tinamit\EnvolturasBF\Tikon\envoltTikon.py -> build\lib\tinamit\EnvolturasBF\Tikon
    copying tinamit\EnvolturasBF\Tikon\__init__.py -> build\lib\tinamit\EnvolturasBF\Tikon
    creating build\lib\tinamit\Interfaz\Imágenes
    copying tinamit\Interfaz\Imágenes\__init__.py -> build\lib\tinamit\Interfaz\Imágenes
    creating build\lib\tinamit\trads\nl
    copying tinamit\trads\nl\BF.py -> build\lib\tinamit\trads\nl
    copying tinamit\trads\nl\Conectado.py -> build\lib\tinamit\trads\nl
    copying tinamit\trads\nl\definiciones.py -> build\lib\tinamit\trads\nl
    copying tinamit\trads\nl\MDS.py -> build\lib\tinamit\trads\nl
    copying tinamit\trads\nl\Modelo.py -> build\lib\tinamit\trads\nl
    copying tinamit\trads\nl\__init__.py -> build\lib\tinamit\trads\nl
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\jmalar1\AppData\Local\Temp\pip-install-imnf4lpi\tinamit\setup.py", line 179, in <module>
        'sdist': dist_f  # Comanda para crear distribución instalable
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages\setuptools\__init__.py", line 129, in setup
        return distutils.core.setup(**attrs)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\core.py", line 148, in setup
        dist.run_commands()
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dist.py", line 974, in run_command
        cmd_obj.run()
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages\setuptools\command\install.py", line 61, in run
        return orig.install.run(self)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\command\install.py", line 545, in run
        self.run_command('build')
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dist.py", line 974, in run_command
        cmd_obj.run()
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\command\build.py", line 135, in run
        self.run_command(cmd_name)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dist.py", line 974, in run_command
        cmd_obj.run()
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages\setuptools\command\build_py.py", line 52, in run
        self.build_packages()
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\command\build_py.py", line 372, in build_packages
        self.build_module(module, module_file, package)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\site-packages\setuptools\command\build_py.py", line 75, in build_module
        package)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\command\build_py.py", line 342, in build_module
        self.mkpath(dir)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\cmd.py", line 338, in mkpath
        dir_util.mkpath(name, mode, dry_run=self.dry_run)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\dir_util.py", line 66, in mkpath
        log.info("creating %s", head)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\log.py", line 44, in info
        self._log(INFO, msg, args)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\distutils\log.py", line 34, in _log
        stream.write('%s\n' % msg)
      File "c:\users\jmalar1\appdata\local\programs\python\python36-32\lib\encodings\cp1252.py", line 19, in encode
        return codecs.charmap_encode(input,self.errors,encoding_table)[0]
    UnicodeEncodeError: 'charmap' codec can't encode characters in position 33-34: character maps to <undefined>

    ----------------------------------------
Command "c:\users\jmalar1\appdata\local\programs\python\python36-32\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\jmalar1\\AppData\\Local\\Temp\\pip-install-imnf4lpi\\tinamit\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().
replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\jmalar1\AppData\Local\Temp\pip-record-0q2wrmex\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\jmalar1\AppData\Lo
cal\Temp\pip-install-imnf4lpi\tinamit\

### PR DESCRIPTION
**Pull request by julienmalard**
## Summary of changes

Fixed regex error in [A-Za-z] test of package names so that the full range of Python's accepted unicode identifiers are accepted.

Closes #1399 (I think).

### Pull Request Checklist
- [X] Changes have tests
- [ ] News fragment added in changelog.d. See [documentation](http://setuptools.readthedocs.io/en/latest/developer-guide.html#making-a-pull-request) for details

### CODE REVIEW

File: pkg_resources/__init__.py [hunk 1]
```diff
@@ -1313,13 +1313,15 @@ def get_default_cache():
...
+
```
Comment (pganssle) [inline pkg_resources/__init__.py]: I really *really* hope this is not the right way to do this. I certainly don't think it should be all on one line.

**Comment by pganssle** [comment]
There appears to be a [canonical regex](https://www.python.org/dev/peps/pep-0508/#names) for package names, unless I'm getting that wrong. I'm not quite sure why we're not using it, though.

It does not seem to be compatible with unicode packages, though, so we should *probably* try and fix this at the spec level.

**Comment by julienmalard** [comment]
Unfortunately it does seem to be the only way the go at this time.
The official spec for identifiers is:

> The identifier syntax is <XID_Start> <XID_Continue>*.
ID_Start is defined as all characters having one of the general categories uppercase letters (Lu), lowercase letters (Ll), titlecase letters (Lt), modifier letters (Lm), other letters (Lo), letter numbers (Nl), the underscore, and characters carrying the Other_ID_Start property. XID_Start then closes this set under normalization, by removing all characters whose NFKC normalization is not of the form ID_Start ID_Continue* anymore.

(From [PEP 3131](https://www.python.org/dev/peps/pep-3131/).)

Unfortunately to use these unicode classes, we have to use the regex, and not the builtin re module. Am I right that adding regex as a dependency to setuptools would be a bad idea?

If so, we are I believe effectively stuck with ugly unicode ranges.

**Comment by pganssle** [comment]
> Unfortunately it does seem to be the only way the go at this time.
> The official spec for identifiers is:

That is for identifiers, this is for package names and versions. It's not clear to me that they are the same (though there's some case to be made that they should be harmonized in the official spec).

> Unfortunately to use these unicode classes, we have to use the regex, and not the builtin re module. Am I right that adding regex as a dependency to setuptools would be a bad idea?
> 
> If so, we are I believe effectively stuck with ugly unicode ranges.

Well, the other option is to not use a regular expression, for example:

"

**Comment by pganssle** [comment]
> Unfortunately it does seem to be the only way the go at this time.
> The official spec for identifiers is:

That is for identifiers, this is for package names and versions. It's not clear to me that they are the same (though there's some case to be made that they should be harmonized in the official spec).

> Unfortunately to use these unicode classes, we have to use the regex, and not the builtin re module. Am I right that adding regex as a dependency to setuptools would be a bad idea?
> 
> If so, we are I believe effectively stuck with ugly unicode ranges.

Well, the other option is to not use a regular expression, for example, this will cover most of what your regex does. Presumably `unicodedata` can also get the Other_ID_start property as well.

```python
def fc(x):
    out = ""
    for c in x:
        if (
            unicodedata.category(c) in {"Lt", "Lm", "Lu", "Ll", "Lt", "Lm", "Lo", "NI"}
            or c == "_"
        ):
            out += c
        else:
            out += "-"
    return re.sub("-+", "-", out).strip("-")
```

''.join(

**Comment by jaraco** [comment]
> Am I right that adding regex as a dependency to setuptools would be a bad idea.

I'd like to think not, but it's not as simple as adding a dependency in another project. Any dependency (and its dependencies) must be vendored. This shouldn't be done with abandon, as it does impact downstream system packagers that work to unvendor the vendored packages.

**Comment by julienmalard** [comment]
Thank you for the feedback! I think I will try out @pganssle 's approach first. By the way, pardon my ignorance, but what is the practical difference between the package "safe name" and its identifier in terms of how Python (or setuptools) uses them?

**Comment by pganssle** [comment]
@julienmalard The two things are nearly entirely different. For something to be a valid identifier means that it can be used as a variable name in Python. For something to be a valid package name means that it can be used for the name of a package to be installed. For example `my-package` is a valid package name, but not a valid identifier, and in fact `my-package`, `my_package` and `My-Package` all refer to the same package (`my-package` is the canonical name).

**Comment by julienmalard** [comment]
Thank you @pganssle for your clear explanation!
I have implemented your alternative to the regex as well and it seems to be passing all tests now.

Please let me know what you think!

File: pkg_resources/__init__.py [hunk 2]
```diff
@@ -1309,17 +1314,40 @@ def get_default_cache():
...
     """
     return (
-        os.environ.get('PYTHON_EGG_CACHE')
-        or appdirs.user_cache_dir(appname='Python-Eggs')
+            os.environ.get('PYTHON_EGG_CACHE')
+            or appdirs.user_cache_dir(appname='Python-Eggs')
     )
 
 
+def fc(x, replace_with='-', also_accept=None, also_refuse=None):
```
Comment (jaraco) [inline pkg_resources/__init__.py:1322]: This function really needs a docstring, especially because `fc` doesn't mean anything to me.

**Comment by jaraco** [comment]
I started down the path of removing the stylistic changes, and re-organizing the commits into first a test failure. However, I noticed that `safe_name` isn't implicated in the traceback of the original report, so I'm not even confident the additional test here exhibits the failure reported in the issue.

It's not obvious to me that the reported issue is even related to the fix herein. Let's go back to the issue and work on creating a minimal reproducer that exhibits the failure.

**Pull request closed without merging**